### ISSUE TITLE
Feature: add support to be able to specify base-16 and optionally base-256 colors

### ISSUE THREAD

**Issue by reportaman** [akinsho/bufferline.nvim#326]
Using base-16 colors have an advantage over using hex RGB colors in the sense that it allows users to be able to change the colors on the fly easily as terminal color theme changes.

The idea is pretty simple: if apps are limited to use colors from a pool of 16 colors, given those colors are inherited from the underlying virtual terminal program, if the program allows changing the color theme on the fly then app colors adapt to the new color theme without users having to any command. For instance, if your virtual terminal has a feature to automatically change the color theme by time of the day, the colors in nvim and other apps will change if the apps limit themselves to using colors 0-15.

While you would better know how the interface change would look like, a way could be as shown below. Bufferline would check ``termguicolors`` to decide whether to apply hex colors or base-16 colors.

```
    require'bufferline'.setup {
        highlights = {
            fill = {
                guifg = '#777777',
                guibg = '#252525',
                ctermfg = 1, // <---- new field
                ctermbg = 7, // <---- new field
                },
```

**Comment by akinsho** [comment]
@reportaman so this is a much bigger request than I think it might seem. I actually purposely chose not to support base-16 colours when I first developed this plugin because I needed to be able to manipulate colours, e.g. lighten and darken colours. I have no idea how to do this with `cterm` colours. At this point though, given that the plugin is built around hex colours it would take solving the problem of being able to manipulate `cterm` colours in the same way that we do hex colours and then finding some way to interchangeably use `cterm` colours without disrupting the use of base 256 for everyone else.

To be clear *I personally do not intend to do this,* it is likely to be a good chunk of work that I don't personally have time for and will provide no benefit to me at all. As always with OSS, if you are willing to implement this yourself and can provide a solution that 

1) most importantly, does not disrupt everyone else's usage of this plugin
2) doesn't introduce bugs or large amounts of complexity

Then I'd be happy to review and merge such a solution.
